src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -158,9 +158,16 @@ public function replace($messages, $domain = 'messages')
     public function add($messages, $domain = 'messages')
     {
         if (!isset($this->messages[$domain])) {
-            $this->messages[$domain] = $messages;
-        } else {
-            foreach ($messages as $id => $message) {
+            $this->messages[$domain] = [];
+        }
+        $intlDomain = $domain;
+        if (false === mb_strpos($intlDomain, self::INTL_DOMAIN_SUFFIX)) {
+            $intlDomain .= self::INTL_DOMAIN_SUFFIX;
+        }
+        foreach ($messages as $id => $message) {
+            if (isset($this->messages[$intlDomain]) && \array_key_exists($id, $this->messages[$intlDomain])) {
```

Comment:
why not use `isset($this->messages[$intlDomain][$id])`? can the value really be null?

Reply:
yes, null values are possible (at least when extracting messages using php-translation)